### PR TITLE
fix(netrw): patch navigation

### LINKED ISSUE #24721: Navigating back from netrw results in "Mark has invalid line number"
### Problem

When I navigate to a file, open `Netrw` and click `ctrl+o` to go back to the file it prints "Mark has invalid line number". I am just getting started with Netrw instead of a normal filetree plugin and maybe this is expected but couldn't find anything about it.

### Steps to reproduce

1. nvim .
2. pick a random file
3. `:Ex`
4. `ctrl+o
5. "Mark has invalid line number"

### Expected behavior

1. nvim .
2. pick a random file
3. `:Ex`
4. `ctrl+o`
5. Comes back to the random file from before

### Neovim version (nvim -v)

NVIM v0.10.0-dev-839+g7aad4643f

### Vim (not Nvim) behaves the same?

using defaults:
no, 9.0.1544 (:Explore not an editor command)

using `vim` normally it's doing the same as in Neovim

### Operating system/version

macOS 13.5

### Terminal name/version

Iterm 3.4.19 (also tried the default MacOS terminal)

### $TERM environment variable

xterm-256color

### Installation

build from repo

### PR DESCRIPTION
**Pull request by lsvmello**
# Problem

Navigation marks are not properly used after opening a file through `netrw`.

# Solution

Patched the `@*` and `@+` logic from Vim's `netrw`.

--- 

I wasn't able to find the exact commit that this was introduced that's why I didn't create this as a patch commit.

Fixes #24721.

### CODE REVIEW

**Comment by lsvmello** [comment]
> Does this problem also exist in vim? If so, mind sending the PR there instead? I think we want to differ as little as possible when it comes to these builtin plugins.

No, it works on Vim.
All I did was take Vim's code into Neovim's code.

**Comment by clason** [comment]
We should be fully up to date on Vim's runtime files (except the ones that were rewritten in Vim9script). Can you check when these changes were added in Vim? If we're missing (part of) a runtime file update, we need to port that (using the same format as other updates so our tooling won't get confused).

File: runtime/autoload/netrw.vim [hunk 1]
```diff
@@ -702,6 +702,11 @@ fun! netrw#Explore(indx,dosplit,style,...)
   endif
 
   " save registers
+  if has("clipboard") && g:netrw_clipboard
+"   call Decho("(netrw#Explore) save @* and @+",'~'.expand("<slnum>"))
+   sil! let keepregstar = @*
+   sil! let keepregplus = @+
```
Comment (justinmk) [inline runtime/autoload/netrw.vim:708]: These `let @+ = ...` calls were intentionally removed in https://github.com/neovim/neovim/commit/6c22c7ab97cca9f8dda6863ee7f1db1ce30a3451 for performance.

netrw is super noisy about writing to the clipboard, which is slow in Nvim. 

Nvim clipboard could be made faster with an approach like https://github.com/neovim/neovim/issues/11804#issuecomment-1113927971

Comment (dundargoc) [reply runtime/autoload/netrw.vim:708]: Ah, I see. Do you think vim would be interested in this patch?

**Pull request closed without merging**